### PR TITLE
Refs [TASK][36139] Tạo UI menu app

### PR DESCRIPTION
**Pull request by hhd0307**
## Related Tickets
- [#Task 36139: Tạo UI menu app](https://edu-redmine.sun-asterisk.vn/issues/36139)
## WHAT
- Tạo giao diện menu app
- Tạo toolbar
## Evidence (Screenshot or Video)
Screenshot_1620985865
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/menu/menu_bottom.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:id="@+id/menu_champion"
```
Comment (tuanlvt-1117) [inline app/src/main/res/menu/menu_bottom.xml]: convention. id type kotlin

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -4,6 +4,40 @@
...
+        android:textAlignment="center"
+        app:titleMarginStart="50dp"
+        app:titleTextColor="@android:color/white"
+        android:background="@android:color/black"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <FrameLayout
+        android:id="@+id/frame_container"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/activity_main.xml]: convention

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -4,6 +4,40 @@
...
     android:layout_height="match_parent"
+    android:fitsSystemWindows="true"
     tools:context=".ui.MainActivity">
 
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbarTop"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textAlignment="center"
+        app:titleMarginStart="50dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/java/com/sun/hero_01/ui/class/ClassFragment.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.sun.hero_01.ui.`class`
+
+import android.view.View
+import com.sun.hero_01.R
+import com.sun.hero_01.base.BaseFragment
+
+class ClassFragment : BaseFragment() {
+
+    override var bottomNavigationViewVisibility = View.VISIBLE
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/class/ClassFragment.kt]: sao không bỏ vào trong base fragment luôn e.
case nào cần thì mới gọi ra

File: app/src/main/java/com/sun/hero_01/ui/MainActivity.kt [hunk 5]
```diff
@@ -2,15 +2,63 @@ package com.sun.hero_01.ui
...
+                else -> false
+            }
+        }
+    }
+
+    private fun loadFragment(fragment: Fragment) {
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.replace(R.id.frame_container, fragment)
+        transaction.addToBackStack(null)
+        transaction.commit()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/MainActivity.kt]: scope function

File: app/src/main/java/com/sun/hero_01/base/BaseFragment.kt [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.Fragment
+import com.sun.hero_01.ui.MainActivity
+
+
+abstract class BaseFragment : Fragment() {
+
+    protected open var bottomNavigationViewVisibility = View.VISIBLE
+
+    abstract fun getLayoutId(): Int
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/hero_01/base/BaseFragment.kt]:
```suggestion
    abstract val layoutResourceId: Int
```

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**